src/js/pages/room/index.js

```diff
@@ -529,10 +529,37 @@ const Settings = ({open}) => {
   // Hooks
   //
   const [screen, setScreen] = useState(SETTINGS_SCREEN_MENU);
+  const [codeCopied, setCodeCopied] = useState(false);
+  const inviteCode = useSelector(BuoySelectors.connectedBuoy).get('token');
+
+  ////
+  // Action callbacks
+  //
+  const copyInviteCode = () => {
+    const dummy = document.createElement('textarea');
+    dummy.value = inviteCode;
+    document.body.appendChild(dummy);
+    dummy.select();
+    document.execCommand('copy');
+    document.body.removeChild(dummy);
+
+    setCodeCopied(true);
+    setTimeout(() => {
+      setCodeCopied(false);
+    }, 2000);
```

Comment:
One thing to be cautious of with using `setTimeout` in React (and especially with React hooks). If the user were to leave this settings screen, the component would be unmounted but this timeout would still be ticking in the background. When it eventually fired, it would call `setCodeCopied`, which would be mutating the state of a component which no longer existed (which I believe will throw an error, but I'm not 100% sure with react hooks).

Unfortunately timeouts are a [bit complicated with hooks](https://overreacted.io/making-setinterval-declarative-with-react-hooks/) but I've incorporated a helper in `utils/react.js` which should help. It's using an interval rather than a timeout, but you should be able to either a) adapt it to use timeouts by following that blog post, or b) finnangle this use case into an interval which you then clear.

It's kind of a pain in the ass, but unfortunately is more or less the way things work in the react world. If we weren't using hooks you'd do this with something like this:

```javascript
class SomeComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {text: 'wat'};
  }

  componentDidMount() {
    this._timeout = setTimeout(() => this.setState({text: 'ok'});
  }

  componentWillUnmount() {
     if (this._timeout) {
      clearTimeout(this._timeout);
    }
  }

  render() {
    return <div>{this.state.text}</div>
  }
}
```

Reply:
Thanks for the info and the blog post :+1:
It seems like I would have a hard time kicking off the `useInterval` from within a function in the component. I get:
`Invalid hook call. Hooks can only be called inside of the body of a function component.`
for:
```javascript
const copyInviteCode = () => {
 // <clipboard stuff>
  useInterval(() => {
    setCodeCopied(false);
  }, 2000);
};
```

Might be best to just keep the copied state and not reset it. I'm noticing though that if I don't reset `codeCopied`, leaving and re-entering the settings screen doesn't revert the link back to its original state. That makes me think maybe the screen doesn't unmount when you leave? Another thing that makes me think that is the old code correctly updates the link even after leaving the screen, and doesn't throw an error.

I guess to figure this out I'd need to find a way to make sure the component is really unmounting?